From fb08c34eee2c883a01ab0dda2a2e34a290516a2a Mon Sep 17 00:00:00 2001
From: Jeremy Evans <code@jeremyevans.net>
Date: Fri, 17 Jul 2015 11:31:45 -0600
Subject: [PATCH 5/8] Use OP_NO_TICKET when testing SSL session cache callbacks

This fixes the test when using LibreSSL and possibly some
configurations of OpenSSL.

diff --git test/openssl/test_ssl_session.rb test/openssl/test_ssl_session.rb
index d4713d9..58fa20b 100644
--- test/openssl/test_ssl_session.rb
+++ test/openssl/test_ssl_session.rb
@@ -316,6 +316,7 @@ __EOS__
 
     ctx_proc = Proc.new { |ctx, ssl|
       ctx.session_cache_mode = OpenSSL::SSL::SSLContext::SESSION_CACHE_SERVER
+      ctx.options = OpenSSL::SSL::OP_NO_TICKET
       last_server_session = nil
 
       # get_cb is called whenever a client proposed to resume a session but
-- 
2.4.5

